src/precord.c

```diff
@@ -264,12 +259,12 @@ UInt FindPRec( Obj rec, UInt rnam, UInt *pos, int cleanup )
             /* We are not allowed to cleanup, so we live with it, we
              * first try to find rnam in the mess at the end, then
              * fall back to binary search: */
-            i = high;
+            UInt i = high;
             while (i >= 1) {
-                rnam2 = (Int) (GET_RNAM_PREC(rec,i));
+                Int rnam2 = (Int)(GET_RNAM_PREC(rec, i));
                 if (rnam == rnam2) {
-                    *pos = i;
-                    return 1;
+                    GAP_ASSERT(i != 0);
```

Comment:
Why this assertion? The loop condition proves that `i >= 1` holds, hence `i != 0` always must be satisfied.

Reply:
It was more just a silly brute-force, I put those in at all return values which have to return a non-zero value, as the rnam was found.